sycl/include/sycl/types.hpp

```diff
@@ -778,7 +792,8 @@ template <typename Type, int NumElements> class vec {
 
   template <typename Ty = DataT>
   explicit constexpr vec(const EnableIfNotHostHalf<Ty> &arg)
-      : m_Data{(DataType)vec_data<Ty>::get(arg)} {}
+      : m_Data{DataType(vec_data<Ty>::get(arg))} {
+  }
```

Comment:
I think this will still initialize a `m_Data` with -1 if given `true` as input - the argument given to initialize a `bool` ext_vector should be casted to something the special behavior of splatting a bool.

Reply:
Also, the `operator=` overload right below this function will still have the same bool splatting behavior.

Reply:
For `sycl::vec<bool, N>`,  If `Ty = bool`, then following declarations suggest that `vec_data<Ty>::get` is a `bool`:
```c++
template <typename T> T vec_helper<T>::get();
template <typename T> using vec_data = detail::vec_helper<T>;

Reply:
> @maarquitos14, can you add a test case to show that the `sycl::vec<bool, N>` operations are correct with this change?

I have just added a file with many tests, but please, feel free to suggest new ones if you think they should be added.


> Also, the `operator=` overload right below this function will still have the same bool splatting behavior.
I believe it's not necessary to change the `operator=` since the rhs should have correct values already. I might be wrong, though, so if you find a case where this could be an issue, please, share it with me so that I can add it to the test file and also update the code accordingly.

Reply:
> Right, that makes sense. And I realized I didn't see/forgot about the specialization added for vec_helper so that's why I thought this wouldn't work.

Ah, right, that is indeed the key to make this work.